Objects/templateobject.c

```diff
@@ -22,6 +22,11 @@ templateiter_next(templateiterobject *self)
     PyObject *item;
     if (self->from_strings) {
         item = PyIter_Next(self->stringsiter);
+        if (PyUnicode_GET_LENGTH(item) == 0) {
+            Py_DECREF(item);
+            item = PyIter_Next(self->interpolationsiter);
```

Comment:
This is a bit more idiomatic.

```suggestion
            Py_SETREF(item, PyIter_Next(self->interpolationsiter));
```